accepted/2023/NuGetAudit-supress-advisory.md

```diff
@@ -0,0 +1,230 @@
+Therefore, at this time this spec is being written, its implementation is that it makes an HTTP request to every package's Package Details (registration) endpoint, which is slow when there are many packages to check.
+Performance could be improved by changing the implementation to use the new `VulnerabilityInfo` resource, and check for vulnerabilities in the same way that restore (NuGetAudit) does.
+However, this would be a breaking change for package sources that have not implemented `VulnerabilityInfo`.
+On the other hand, we don't know of any package source that provide any package vulnerability info through the registration resource, other than nuget.org.
+
+Finally, `dotnet list package` already reads some properties out of the project, but doesn't currently retrieve any MSBuild items.
+Therefore, some new abstractions or helper methods will probably be needed, and there might not be existing examples in our code to copy-paste, as restore uses different APIs to get the same data.
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+- Priority
+
+The linked issue has few up-votes, so we don't yet have evidence that it's a high impact feature.
+
+- Alternatives
+
+Customers can already achieve a similar outcome, but it will require a lot more effort on their end.
+Firstly, they wouldn't use NuGetAudit, at least on CI.
+Instead, they would use `dotnet list package --format json --vulnerable` (possibly also with `--include-transitive`) as a separate step in their pipeline.
+Finally, they would need to have a script that parses the output, remove vulnerabilities they wish to suppress, then return a non-zero exit code if the result still contains any other vulnerabilities.
+
+This effort would need to be duplicated by every customer who wishes to implement such behavior.
+Even if someone in the community (or even the NuGet team ourselves) provides a sample, it requires developers to find the location where the sample is documented, and then to adapt it to their needs.
+Alternatively, if this feature is implemented, then it's much easier for customers to adopt, as it's a fairly small configuration change.
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+
+### MSBuild vs nuget.config
+
+Some NuGet features are only available via *nuget.config*, and not MSBuild items or properties.
+However, we had [one customer provide feedback that they'd like the suppression to be per-project](https://github.com/NuGet/Home/issues/11926#issuecomment-1683384471), rather than per repo.
+This allows a project already using a package with a known vulnerability to prevent CI pipeline errors, while preventing other projects in the solution from starting to use it.
+This seems like an important use-case that should be enabled.
+
+### MSBuild item vs property
```

Comment:
For perf/agility reasons my preference is a property, but I think we need customer validation for that. 

Going from property to item is doable, but going from item to property would cause more issues for customers.

Reply:
We eventually decided on items over properties, primarily because they allow for metadata, and will make it easier to add features later on, but I did make a note of the perf concerns in the spec.